src/routes/config.ts

```diff
@@ -1,33 +1,47 @@
 import { RequestHandler } from 'express';
 import { JSONSchema6 } from 'json-schema';
 import * as _ from 'lodash';
+import * as memoize from 'memoizee';
 
 import {
 	BLACKLISTED_NAMES,
-	HOST_CONFIG_VAR_PROPERTIES,
+	DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES,
 	INVALID_CHARACTER_REGEX,
 	RESERVED_NAMES,
 	RESERVED_NAMESPACES,
-	RESIN_HOST_CONFIG_CAPABLE_DEVICE_TYPES,
 	SUPERVISOR_CONFIG_VAR_PROPERTIES,
 	WHITELISTED_NAMES,
 	WHITELISTED_NAMESPACES,
 } from '../lib/env-vars';
 
+const getVarsPropsForDeviceType = memoize(
+	(deviceType?: string) => {
+		if (!deviceType) {
+			return SUPERVISOR_CONFIG_VAR_PROPERTIES;
+		}
+
+		return Object.assign(
+			{},
+			SUPERVISOR_CONFIG_VAR_PROPERTIES,
+			...DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES.filter(config =>
+				config.capableDeviceTypes.includes(deviceType),
+			).map(config => config.properties),
+		);
+	},
+	{
+		maxAge: 5 * 60 * 1000,
```

Comment:
should this be configurable?

Reply:
Reverted the memoization since that endpoint isn't hot at all (not even warm), so let's keep it simple.
See: https://www.flowdock.com/app/rulemotion/p-opensource/messages/48253